src/app/FakeLib/TargetHelper.fs

```diff
@@ -322,36 +324,114 @@ let listTargets() =
 // Instead of the target can be used the list dependencies graph parameter.
 let doesTargetMeansListTargets target = target = "--listTargets"  || target = "-lt"
 
+/// Determines a parallel build order for the given set of targets
+let determineBuildOrder (d : TargetTemplate<unit>) =
```

Comment:
please add unit tests. Sure, they don't prove anything about the function, but experience shows they work very well in practice. Don't forget we have more than 100 contributors (and much more users) from very different fields (a lot don't have any computer science background). Having a set of tests which checks basic edge cases will make it much easier for future contributors to maintain this function. 

If you still think unit tests are useless here, then please look at [FsCheck](https://fsharp.github.io/FsCheck/) and try to prove some important properties. We already have a FsCheck project at https://github.com/fsharp/FAKE/tree/master/src/test/FsCheck.Fake.